src/Illuminate/Database/Eloquent/Relations/Concerns/ComparesRelatedModels.php

```diff
@@ -64,7 +64,10 @@ abstract protected function getRelatedKeyFrom(Model $model);
      */
     protected function compareKeys($parentKey, $relatedKey)
     {
-        if (empty($parentKey) || empty($relatedKey)) {
+        if ($parentKey === null
+            || $parentKey === ''
+            || $relatedKey === null
+            || $relatedKey === '') {
```

Comment:
Did you consider to use the `blank()` helper? 

I guess these checks are fine, but `blank()` would keep it shorter and more legible, in my opinion

Reply:
I did not know that blank existed. I agree with you. :)